app/health/select.go

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package health
+
+import (
+	"regexp"
+
+	pb "github.com/prometheus/client_model/go"
+)
+
+type labelSelector func(*pb.MetricFamily) *pb.Metric
+
+func maxLabel(metricsFam *pb.MetricFamily) *pb.Metric {
+	var max *pb.Metric
+	for _, metric := range metricsFam.Metric {
+		if max == nil {
+			max = metric
+		}
+
+		var val float64
+		switch metricsFam.GetType() {
+		case pb.MetricType_COUNTER:
+			val = metric.Counter.GetValue()
+		case pb.MetricType_GAUGE:
+			val = metric.Gauge.GetValue()
+		default:
+			panic("invalid metric type for simple value labelSelector")
+		}
+
+		if val > max.Counter.GetValue() {
+			max = metric
+		}
```

Comment:
how do you know that `max` metric will always be a counter metric? i think it depends on `metric` variable of the for loop which gets assigned to max in this if block

Reply:
maybe it is better to have two variables: `maxValue` and `maxMetric`. `maxValue` compares the values of the metrics and `maxMetric` stores the max metric based on the value.

Reply:
updated